patroni/postgresql/config.py

```diff
@@ -403,6 +404,15 @@ def check_directories(self) -> None:
     def config_dir(self) -> str:
         return self._config_dir
 
+    @property
+    def full_pg_version(self) -> int:
+        if self._postgresql.state == 'running':
+            try:
+                return self._postgresql.server_version
+            except AttributeError:
+                pass
+        return postgres_version_to_int(get_postgres_version(bin_name=self._postgresql.pgcommand('postgres')))
```

Comment:
There is a race condition possible, when major version in PG_VERSION doesn't match the binary.
In this case we better take `self._postgresql.major_version`.